pkg/client/group_mgmt_client.go

```diff
@@ -384,3 +404,58 @@ func unwrapError(body []byte) (string, error) {
 	}
 	return errResp, nil
 }
+
+// processAsyncResponse: process http code 202 response
+func processAsyncResponse(client *GroupMgmtClient, body []byte) (interface{}, error) {
+	errResp, _ := unwrapError(body)
+	if client.WaitOnJob { // check sync flag
+		unwrapMessage := &ErrorResponse{}
+		err := json.Unmarshal(body, unwrapMessage)
+		if err != nil {
+			return nil, err
+		}
+		var jobId string
+		for _, msg := range unwrapMessage.Messages {
+			if strings.Compare(msg.Code, smAsyncJobId) == 0 {
+				jobId = msg.Arguments.JobId
+			}
+		}
+		if len(jobId) == 0 {
+			return nil, fmt.Errorf("http response error: status (202), failed to get the job id")
+		}
+
+		id, err := waitForJobResult(jobId, client)
+		if err != nil {
+			return nil, fmt.Errorf("http response error: status (202), messages: %v", err.Error())
+		}
+		return id, fmt.Errorf("http response error: status (202), messages: %v", errResp)
```

Comment:
Sorry, I didn't understand the response to my prior comment at this line.  If we reach this statement, the job has completed successfully.  Why return an error in addition to the object ID?

Reply:
It should be nil